File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    sumcnt = set()
    n = len(elements)
    elements = elements + elements
    print(elements)
    for i in range (n):
        for j in range (n):
            sumcnt.add(sum(elements[i:i+j]))
    return len(sumcnt)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    var answer = new Set();
    const double = [...elements, ...elements]
    const slidingWindow = (len) => {
        for (let i = 0; i< elements.length; i ++){
            answer.add(double.slice(i,i+len).reduce((acc,cur) => acc +cur, 0))
        }
    }
    
    for (let i = 1; i <= elements.length ; i ++){
        slidingWindow(i)
    }
    return answer.size;
}